### PR TITLE
Fix #46, #53: don't bind to both HTTP and HTTPs.

### LINKED ISSUE #46: Server binds to both HTTPs and HTTP ports?
It seems like the server always binds to both, or just HTTP.

### LINKED ISSUE #53: isPortTaken tests don't work
I put garbage instead of port numbers in the tests that call `isPortTaken` and they still pass. Something looks not right, this should return a promise and reject it?

### PR DESCRIPTION
**Pull request by dblock**
Stop binding to both http and https ports, simplify options, fix #53 and #46.

### CODE REVIEW

**Comment by dblock** [comment]
@tejashah88 Take a look please?

File: README.md [hunk 1]
```diff
@@ -96,7 +96,7 @@ require('alexa-app-server').start({
 	// The directory containing server-side processing modules (see below).
 	server_dir: "server",
 
-	// The port the server should bind to.
+	// The port the server should bind to. Defaults to 8080.
 	port: 80,
```
Comment (tejashah88) [inline README.md]: I'm guessing you meant to write `port: 8080,`, correct?

File: index.js [hunk 2]
```diff
@@ -281,19 +283,16 @@ var appServer = function(config) {
                     self.error("privateKey: '" + config.privateKey + "' or certificate: '" + config.certificate + "' do not exist in /sslcert. HTTPS will not be enabled");
                 }
             } else {
-                self.error("privatekey, httpsPort, or certificate paramater not set in config. HTTPS will not be enabled");
+                self.error("privatekey, or certificate paramater not set in config. HTTPS will not be enabled");
```
Comment (tejashah88) [inline index.js]: Tiny nitpick: remove the comma next to privatekey.

File: README.md (hunk 1 above)
Comment (tejashah88) [inline README.md]: I'm guessing you meant to write `port: 8080,`, correct?

**Pull request merged**